### PR TITLE
CLDC-2571 Add absorbed org relationship

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**
Adds an absorbing_organisation_id to orgs so they can be marked with the org that absorbed them during a merge. As discussed with the team in refinement, we don't need any more complex relationship details so this is enough. Adds the relevant relations to orgs, and includes these absorbed orgs in org filters and the user's log lists.

ticket: https://dluhcdigital.atlassian.net/browse/CLDC-2571

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -192,14 +192,14 @@
     t.integer "hb"
     t.integer "hbrentshortfall"
     t.integer "property_relet"
-    t.datetime "mrcdate", precision: nil
+    t.datetime "mrcdate"
```
Comment (natdeanlewissoftwire) [inline db/schema.rb]: not sure why these are being removed but it doesn't make a difference afaik, default precision either way

File: app/models/form/lettings/questions/stock_owner.rb [hunk 2]
```diff
@@ -23,13 +23,13 @@ def answer_options(log = nil, user = nil)
...
 
-    stock_owners_answer_options = if user.support?
-                                    Organisation.where(holds_own_stock: true)
-                                  else
-                                    user.organisation.stock_owners
-                                  end.pluck(:id, :name).to_h
+    user_answer_options = if user.support?
+                            Organisation.where(holds_own_stock: true)
+                          else
+                            user.organisation.stock_owners + user.organisation.absorbed_organisations
```
Comment (kosiakkatrina) [inline app/models/form/lettings/questions/stock_owner.rb]: What would happen if absorbed organisation doesn't hold own stock?

File: app/models/organisation.rb [hunk 3]
```diff
@@ -1,9 +1,6 @@
 class Organisation < ApplicationRecord
   has_many :users, dependent: :delete_all
   has_many :data_protection_officers, -> { where(is_dpo: true) }, class_name: "User"
-  has_many :owned_lettings_logs, class_name: "LettingsLog", foreign_key: "owning_organisation_id", dependent: :delete_all
-  has_many :managed_lettings_logs, class_name: "LettingsLog", foreign_key: "managing_organisation_id"
-  has_many :owned_sales_logs, class_name: "SalesLog", foreign_key: "owning_organisation_id", dependent: :delete_all
```
Comment (kosiakkatrina) [inline app/models/organisation.rb:6]: Are we not using this anywhere?

Comment (kosiakkatrina) [reply app/models/organisation.rb:6]: nevermind!

File: app/helpers/filters_helper.rb [hunk 4]
```diff
@@ -45,12 +45,12 @@ def selected_option(filter, filter_type)
   end
 
   def owning_organisations_filter_options(user)
```
Comment (kosiakkatrina) [inline app/helpers/filters_helper.rb]: Could we update/add tests for these helper methods?

File: app/models/form/lettings/questions/stock_owner.rb (hunk 2 above)
Comment (natdeanlewissoftwire) [reply app/models/form/lettings/questions/stock_owner.rb]: hm, good point - I'll check for that too and flag at PO, think that's the desired behaviour

File: app/helpers/filters_helper.rb (hunk 4 above)
Comment (natdeanlewissoftwire) [reply app/helpers/filters_helper.rb]: good idea, have done :)

**Review by kosiakkatrina — APPROVED** [review]

**Pull request merged**